geopandas/io/file.py

```diff
@@ -38,18 +40,27 @@ def _is_url(url):
         return False
 
 
-def read_file(filename, bbox=None, **kwargs):
+def read_file(filename, bbox=None, mask=None, data_slice=None, **kwargs):
     """
     Returns a GeoDataFrame from a file or URL.
 
+    .. versionadded:: 0.7.0 mask, data_slice
+
     Parameters
     ----------
     filename: str
         Either the absolute or relative path to the file or URL to
         be opened.
-    bbox : tuple | GeoDataFrame or GeoSeries, default None
-        Filter features by given bounding box, GeoSeries, or GeoDataFrame.
-        CRS mis-matches are resolved if given a GeoSeries or GeoDataFrame.
+    bbox: tuple | GeoDataFrame or GeoSeries | shapely Geometry, default None
+        Filter features by given bounding box, GeoSeries, GeoDataFrame or a
+        shapely geometry. CRS mis-matches are resolved if given a GeoSeries
+        or GeoDataFrame. Cannot be used with mask.
+    mask: dict | GeoDataFrame or GeoSeries | shapely Geometry, default None
+        Filter features by given dict-like geojson geometry, GeoSeries,
+        GeoDataFrame or a shapely geometry. CRS mis-matches are resolved
+        if given a GeoSeries or GeoDataFrame. Cannot be used with bbox.
+    data_slice: slice, default None
+        Load in specific rows by passing in a slice() object.
```

Comment:
Some random thoughts here: 

- doing `data_slice=slice(...)` gives a duplicate "slice" usage. Wondering if we can have another keyword name that avoids this
- I think that it would be nice to support the basic case of reading the first n rows as well (expanding to `slice(n) == slice(None, n)`).

Reply:
> Wondering if we can have another keyword name that avoids this

Have anything in particular in mind? `row_filter`?

Reply:
I kinda like `row_filter` and am going with that one.